### PR TITLE
[IA-5000] Make sure to transition DELETING Azure disks to ERROR

### PR DESCRIPTION
**Pull request by LizBaldo**
__Jira ticket__: https://broadworkbench.atlassian.net/browse/IA-5000

### What

- Ensures that when an Azure runtime deletion fails, both the runtime AND the persistent disk (if provided) transition their statuses to ERROR

### Why

- Even though persistent disks where being deleted properly in the Azure portal, they were still marked as DELETING in the Leonardo database (and therefore the cloud environment page in the UI), leading users to believe that they were leaked resources

### What to test

- 

### Who tested and where

- [ ] This change is covered by automated tests 
  - _NB: Rerun automation tests on this PR by commenting `jenkins retest` or `jenkins multi-test`._
- [ ] I validated this change 
- [ ] Primary reviewer validated this change 
- [ ] I validated this change in the dev environment

### CODE REVIEW

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala [hunk 1]
```diff
@@ -726,6 +728,7 @@ class AzurePubsubHandlerInterp[F[_]: Parallel](
           logger.error(s"Wsm deleteVm job failed due to ${resp.getErrorReport.getMessage}") >> F.raiseError[Unit](
             AzureRuntimeDeletionError(
               params.runtime.id,
+              None,
```
Comment (LizBaldo) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala]: For VM deletion polling, the diskID is not part of the parameters (same for storage container deletion below). If this is just for logging then it is okay, but if raising this error triggers database status changes then I will need to change that.

Comment (lucymcnatt) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala]: hmm yeah I think this is then picked up by the handleAzureRuntimeDeletionError func in the LeoPubsubMessageSubscriber, so we might need these to not be `None`...

Comment (LizBaldo) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala]: done

**Review by mlilynolting — APPROVED** [review]

**Comment by mlilynolting** [comment]
Looks like test "should handle azure vm creation failure properly" is failing?

**Review by lucymcnatt — APPROVED** [review]
LGTM given that one unit test is fixed!

**Pull request merged**